### PR TITLE
refactor so the service needs config and docs are opt-in

### PR DESCRIPTION
**Pull request by erizocosmico**
Instead of having a default owner and an optional mappings file, not the
service needs a config file that specifies the mapping between each single
host that will be available in the service and its repository amonst other
configurations.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,36 +56,44 @@ make build
...
 
-In `/etc/docsrv/mappings.yml` you can optionally put a mappings file, which will contain a map from host to GitHub repos in the following format `${OWNER}/${REPO_NAME}`.
+In `/etc/docsrv/conf.d/config.toml` you need to put the configuration for docsrv, which is a mapping between hosts and project configurations.
+
+Example `config.toml`
 
 ```
-foo.mydomain.tld: myorg/foo
-bar.mydomain.tld: otherorg/bar
+["bar.tomain.tld"]
```
Comment (Serabe) [inline README.md]: Typo or pun?

**Review by Serabe — APPROVED** [review]

File: docsrv/config.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/Masterminds/semver"
+)
+
+// Config is a map from hosts to project configurations.
+type Config map[string]ProjectConfig
+
+// ProjectForHost will returne the owner and repository name of the project
```
Comment (dpordomingo) [inline docsrv/config.go:16]: s/returne/return

File: README.md (hunk 1 above)
Comment (erizocosmico) [reply README.md]: typo lol

**Comment by erizocosmico** [comment]
Amended the typo in the readme

File: docsrv/fetcher.go [hunk 3]
```diff
@@ -71,12 +72,18 @@ func (g *githubFetcher) releases(owner, project string) ([]*release, error) {
 			if release == nil {
 				continue
 			}
+
+			v := newVersion(release.tag)
+			if v != nil && v.LessThan(minVersion) {
```
Comment (dpordomingo) [inline docsrv/fetcher.go]: we should maybe ignore the order, and analyze all versions

**Review by dpordomingo — APPROVED** [review]
LGTM, but I'd consider the following

File: docsrv/fetcher.go (hunk 3 above)
Comment (dpordomingo) [inline docsrv/fetcher.go]: already done, thanks

**Pull request merged**